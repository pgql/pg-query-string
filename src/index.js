import squel from './squel';
export * from './builder';
export default squel;
