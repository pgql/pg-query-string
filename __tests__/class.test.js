import { Table } from '../src';
import cases from 'jest-in-case';

const table = new Table('users', {
  id: 'uuid',
  username: 'text',
  email: 'email',
  password: 'password',
  active: 'boolean',
  github: 'githublink',
  facebook: 'fblink',
  myspace: 'myslink'
});

it('select via class', () => {
  expect(
    table.select(['id', 'email'], {
      username: 'pyramation',
      email: 'pyramation@gmail.com'
    })
  ).toMatchSnapshot();
});

it('selectOne via class', () => {
  expect(
    table.selectOne(['id', 'email'], {
      username: 'pyramation',
      email: 'pyramation@gmail.com'
    })
  ).toMatchSnapshot();
});

it('insert via class', () => {
  expect(
    table.insert({ username: 'pyramation', email: 'pyramation@gmail.com' })
  ).toMatchSnapshot();
});
it('update via class', () => {
  expect(
    table.update(
      {
        username: 'pyramation',
        email: 'pyramation@gmail.com'
      },
      {
        email: 'pyramation@gmail.com'
      }
    )
  ).toMatchSnapshot();
});

it('update many fields via class', () => {
  expect(
    table.update(
      {
        email: 'cheerios@gmail.com',
        username: 'pyramation',
        active: true,
        password: 'secure',
        avatar: 'http://some.link',
        github: 'pyramation',
        launchql: 'yes',
        facebook: 'danlynch',
        myspace: 'forgot',
        tags: 'pyramation'
      },
      {
        email: 'pyramation@gmail.com'
      }
    )
  ).toMatchSnapshot();
});

it('delete via a class', () => {
  expect(
    table.delete({ username: 'pyramation', email: 'pyramation@gmail.com' })
  ).toMatchSnapshot();
  expect(
    table.delete({ username: 'pyramation', email: 'pyramation@gmail.com' }, 2)
  ).toMatchSnapshot();
});
