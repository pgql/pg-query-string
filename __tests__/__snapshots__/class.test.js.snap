// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete via a class 1`] = `
Object {
  "text": "DELETE FROM users WHERE (username = $1::text AND email = $2::email)",
  "values": Array [
    "pyramation",
    "pyramation@gmail.com",
  ],
}
`;

exports[`delete via a class 2`] = `
Object {
  "text": "DELETE FROM users WHERE (username = $1::text AND email = $2::email) LIMIT $3",
  "values": Array [
    "pyramation",
    "pyramation@gmail.com",
    2,
  ],
}
`;

exports[`insert via class 1`] = `
Object {
  "text": "INSERT INTO users (username, email) VALUES ($1::text, $2::email)",
  "values": Array [
    "pyramation",
    "pyramation@gmail.com",
  ],
}
`;

exports[`select via class 1`] = `
Object {
  "text": "SELECT id, email FROM users WHERE (username = $1::text AND email = $2::email)",
  "values": Array [
    "pyramation",
    "pyramation@gmail.com",
  ],
}
`;

exports[`selectOne via class 1`] = `
Object {
  "text": "SELECT id, email FROM users WHERE (username = $1::text AND email = $2::email) LIMIT $3",
  "values": Array [
    "pyramation",
    "pyramation@gmail.com",
    1,
  ],
}
`;

exports[`update many fields via class 1`] = `
Object {
  "text": "UPDATE users SET email = $1::email, username = $2::text, active = $3::boolean, password = $4::password, avatar = $5, github = $6::githublink, launchql = $7, facebook = $8::fblink, myspace = $9::myslink, tags = $10 WHERE (email = $11::email)",
  "values": Array [
    "cheerios@gmail.com",
    "pyramation",
    true,
    "secure",
    "http://some.link",
    "pyramation",
    "yes",
    "danlynch",
    "forgot",
    "pyramation",
    "pyramation@gmail.com",
  ],
}
`;

exports[`update via class 1`] = `
Object {
  "text": "UPDATE users SET username = $1::text, email = $2::email WHERE (email = $3::email)",
  "values": Array [
    "pyramation",
    "pyramation@gmail.com",
    "pyramation@gmail.com",
  ],
}
`;
